jetty-home/src/main/resources/bin/jetty.sh

```diff
@@ -503,13 +526,27 @@ fi
 ##################################################
 case "$ACTION" in
   start)
-    echo -n "Starting Jetty: "
-
     if (( NO_START )); then
       echo "Not starting ${NAME} - NO_START=1";
       exit
     fi
 
+    if ! touch "$JETTY_PID"
```

Comment:
Assumes that `touch` is available.

Can't we use `bash` built-in stuff like we do when testing for readability?
I.e. `if [ ! -w "$JETTY_PID" ]`

Reply:
The benefit of using `touch` is that if the file doesn't exist it will attempt to create it, which will trigger the permission issue (if such an issue exists)
The `-w` bash option will only return "true if file exists and is writable"

```
# Lets try testing a file that exists, is writable, but I don't have permissions to write.
[joakim@hyperion ~]$ ls -la /var/run/ubuntu-fan/
total 0
drwxr-xr-x  2 root root   60 Sep 28 07:36 .
drwxr-xr-x 46 root root 1340 Sep 28 07:36 ..
-rw-r--r--  1 root root    0 Sep 28 07:36 .lock
[joakim@hyperion ~]$ if [ -w /var/run/ubuntu-fan/.lock ] ; then echo "Is writable"; else echo "Is NOT writable"; fi
Is NOT writable

# Now lets try a bogus file in a directory I'm not allowed to write in
[joakim@hyperion ~]$ if [ -w /var/run/ubuntu-fan/bogus ] ; then echo "Is writable"; else echo "Is NOT writable"; fi
Is NOT writable

# Now lets try a bogus file in a directory I am ALLOWED to write in
[joakim@hyperion ~]$ if [ -w bogus ] ; then echo "Is writable"; else echo "Is NOT writable"; fi
Is NOT writable

# Now lets try a real file, that exists, and I can write to
[joakim@hyperion ~]$ if [ -w .bashrc ] ; then echo "Is writable"; else echo "Is NOT writable"; fi
Is writable
```

Reply:
Also, `touch` is not a function I would worry about.

We've had it in `jetty.sh` for a long time without issue.
https://github.com/jetty-project/codehaus-jetty6/blob/master/bin/jetty.sh#L541

Reply:
Okay.